src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.cpp

```diff
@@ -654,6 +654,23 @@ namespace UiaOperationAbstraction
         return m_useRemoteApi;
     }
 
+    UiaCacheRequest UiaOperationDelegator::CreateCacheRequest()
+    {
+        if (m_useRemoteApi)
+        {
+            return m_remoteOperation.NewCacheRequest();
+        }
+        else
+        {
+            winrt::com_ptr<IUIAutomation> automation;
+            THROW_IF_FAILED(::CoCreateInstance(__uuidof(CUIAutomation8), nullptr, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(automation.put())));
+
+            winrt::com_ptr<IUIAutomationCacheRequest> cacheRequest;
+            automation->CreateCacheRequest(cacheRequest.put());

```

Comment:
>CreateCacheRequest [](start = 24, length = 18)

`THROW_IF_FAILED`

Reply:
@mwcampbell 
> In the abstraction, I don't think the definition of `UiaCacheRequest` belongs in the generated code, because it can be completely hand-written. I think I know why it's defined in the generated header file; it depends on the enum types that are defined in that header file. But I think this just means we need a separate generated fragment for the enum definitions, which can be included right after the definition of `UiaEnum` itself. Then, if you define `UiaCacheRequest` in UiaOperationAbstraction.h, you don't have to make the code generator write that definition, and you can add usability improvements like a default constructor.

I created it with the code generator because it provides boilerplate like IsNull and operators. Looking more closely at other types, though, you're probably right that it differs enough it's easier to hand-write it.

Reply:
Huh, how did this get here? Weird.

Anyway, @mwcampbell, this is meant to reply to the quoted comment lol.